### ISSUE TITLE
weird log_prob values returned by ar_log_likelihood

### ISSUE THREAD

**Issue by r-shruthi11** [dattalab/jax-moseq#26]
Calling `ar_log_likelihood` on `x` produces values that are >0. Log probabilities should be upper bounded at 0. 

```
x = np.copy(results[key]['latent_state'])
print(x.shape)


Ab = np.copy(checkpoint['params']['Ab'])
print(Ab.shape)


Q = np.copy(checkpoint['params']['Q'])
print(Q.shape)

from jax_moseq.utils.autoregression import get_nlags, ar_log_likelihood
log_probs = ar_log_likelihood(x, (Ab[0], Q[0]))
print(log_probs.min())
print(log_probs.max())
```

<img width="1232" alt="Screen Shot 2023-07-19 at 1 31 24 PM" src="https://github.com/dattalab/jax-moseq/assets/31861615/c886e945-80c0-4a12-814a-ad33801e928e">

**Comment by calebweinreb** [comment]
The log probabilities don't have to be negative since they represent values of a probability density function... Think about e.g. the density function for a univariate normal -- even though the whole thing integrates to 1 it can still be >1 at its peak.

**Comment by r-shruthi11** [comment]
Hmm, I'm trying to extract cross-syllable likelihoods similar to this figure in your preprint <img width="1232" alt="Screen Shot 2023-07-19 at 1 31 24 PM" src="https://github.com/dattalab/jax-moseq/assets/31861615/c886e945-80c0-4a12-814a-ad33801e928e"> and it appears as if the the probabilities and the corresponding log probabilities are bounded.

**Comment by calebweinreb** [comment]
They're bounded because the cross-likelihoods are normalized, something like: 

cross_likelihood[i,j] = P(frames assigned as state j | dynamics of state i) / P(frames assigned as state j | dynamics of state j)

The formal description is on page 33 of the supplement of the original MoSeq paper. Here's a function that calculates it:

```

def state_cross_likelihoods(params, states, mask, **kwargs):
    
    x, Ab, Q = jax.device_put((states['x'],params['Ab'],params['Q']))
    log_likelihoods = jax.lax.map(partial(ar_log_likelihood, x), (Ab, Q))
    
    nlags = mask.shape[1] - log_likelihoods.shape[2]
    log_likelihoods = np.moveaxis(log_likelihoods,0,2)[mask[:,nlags:]>0]

    z = states['z'][mask[:,nlags:]>0]
    changepoints = np.diff(z).nonzero()[0]+1
    counts = np.bincount(z[changepoints])
    
    n_states = log_likelihoods.shape[1]
    cross_likelihoods = np.zeros((n_states,n_states))
    for j in range(n_states):
        ll = log_likelihoods[z==j].sum(0)
        cross_likelihoods[j] = (ll-ll[j])/(counts[j] + 1e-6)
    return cross_likelihoods
 ```
